service/src/main/java/org/cbioportal/genome_nexus/service/cached/BaseCachedVariantAnnotationFetcher.java

```diff
@@ -0,0 +1,102 @@
+import org.cbioportal.genome_nexus.service.ResourceTransformer;
+import org.cbioportal.genome_nexus.service.exception.ResourceMappingException;
+import org.cbioportal.genome_nexus.util.NaturalOrderComparator;
+import org.springframework.dao.DataAccessResourceFailureException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class BaseCachedVariantAnnotationFetcher
+    extends BaseCachedExternalResourceFetcher<VariantAnnotation, VariantAnnotationRepository>
+{
+    private static final Log LOG = LogFactory.getLog(BaseCachedVariantAnnotationFetcher.class);
+
+    protected String collection;
+    protected VariantAnnotationRepository repository;
+    protected Class<VariantAnnotation> type;
+    protected ExternalResourceFetcher<VariantAnnotation> fetcher;
+    protected ResourceTransformer<VariantAnnotation> transformer;
+    protected Integer maxPageSize;
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer)
+    {
+        super(collection, repository, type, fetcher, transformer, Integer.MAX_VALUE);
+    }
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer,
+                                             Integer maxPageSize)
+    {
+        super(collection, repository, type, fetcher, transformer, maxPageSize);
+    }
+   
+    @Override
+    protected String extractId(VariantAnnotation instance)
+    {
+        return instance.getVariantId();
+    }
+
+    @Override
+    protected String extractId(DBObject dbObject)
+    {
+        return (String)dbObject.get("input");
+    }
+
+    @Override 
+    public VariantAnnotation fetchAndCache(String id) throws ResourceMappingException
+    {
+        VariantAnnotation variantAnnotation = null;
+        try {
+            variantAnnotation = super.fetchAndCache(id);
+            if (variantAnnotation == null) {
+                variantAnnotation = new VariantAnnotation(id);
+            } else {
+                variantAnnotation.setSuccessfullyAnnotated(true);
```

Comment:
so is it necessary to override this value if the model has some logic for resolving the value in the constructor? Or maybe we can move the logic to the getter?

Reply:
I think the problem was since we're casting the response to VariantAnnotation type and the response does not naturally have a successfully annotated field, it does not get properly set.